src/logic/Utils.ts

```diff
@@ -186,16 +186,41 @@ export const generateSchemesOptions = (schemes: SchemeIni) => {
   ));
 };
 
-// Reset any Marketplace localStorage keys (effectively resetting it completely)
-export const resetMarketplace = () => {
+/**
+ * Reset Marketplace localStorage keys
+ * @param category The category to reset. If not provided, reset all categories.
+ */
+export const resetMarketplace = (category: "extensions" | "snippets" | "theme" | "all" = "all") => {
   console.debug("Resetting Marketplace");
 
-  // Loop through and reset marketplace keys
-  Object.keys(localStorage).forEach((key) => {
-    if (key.startsWith("marketplace:")) {
-      localStorage.removeItem(key);
-      console.debug(`Removed ${key}`);
-    }
+  const keysToRemove: string[] = [];
+
+  // If have category, just reset that category
+  if (category === "extensions") {
+    // Remove the extensions themselves
+    keysToRemove.push(...getLocalStorageDataFromKey(LOCALSTORAGE_KEYS.installedExtensions, []));
+    // Remove the list of extension keys
+    keysToRemove.push(LOCALSTORAGE_KEYS.installedExtensions);
+  } else if (category === "snippets") {
+    keysToRemove.push(...getLocalStorageDataFromKey(LOCALSTORAGE_KEYS.installedSnippets, []));
+    keysToRemove.push(LOCALSTORAGE_KEYS.installedSnippets);
+  } else if (category === "theme") {
+    keysToRemove.push(...getLocalStorageDataFromKey(LOCALSTORAGE_KEYS.installedThemes, []));
+    keysToRemove.push(LOCALSTORAGE_KEYS.installedThemes);
+    keysToRemove.push(LOCALSTORAGE_KEYS.themeInstalled);
```

Comment:
Why do we have 2 of these lol

Reply:
Because it has the standard array like the others use, and then it's got a second key that just has the one, since the app references the active theme in a number of places. We can refactor that later if we want; I don't feel like changing how it works in this PR.

Reply:
That makes sense yeah.